core/src/main/java/io/ddf/util/Utils.java

```diff
@@ -66,6 +69,46 @@ public static DataFormat getDataFormatFromPath(String path) throws DDFException
     }
   }
 
+
+  public static List<String> sanitizeColumnName(List<String> columnNames) throws DDFException {
+    Preconditions.checkArgument(columnNames.size() > 0);
+    List<String> ret = new ArrayList<>();
+    for (String name : columnNames) {
+      Preconditions.checkArgument(name.length() > 0);
+      String dashName = name.trim().replaceAll("![0-9a-zA-Z]|\\s", "_");
+      int i;
+      for (i = 0; i < dashName.length(); ++i) {
+        char c = name.charAt(i);
+        boolean isAlpha = ((c >= 'a' && c <= 'z') || (c >= 'A' && c <='Z'));
+        if (isAlpha) {
+          ret.add(dashName.substring(i));
+          break;
+        }
+      }
+      if (i == dashName.length()) {
+        // TODO: get some random name
+        throw new DDFException(String.format("Not valid column name: %s", name));
+      }
+    }
+    HashSet<String> nameSet = new HashSet<>();
```

Comment:
Right, not just this piece of code, it's good if we can document what this function does in the beginning of this function.